src/NuGet.Core/NuGet.Protocol/Utility/ValidatePackageId.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using NuGet.Common;
+
+namespace NuGet.Protocol
+{
+    internal class ValidatePackageId
```

Comment:
>ValidatePackageId [](http://example.com/codeflow?start=19&length=17)

It would be good if the PR description could explain why this class is necessary in the NuGet.Protocol namespace. i.e. I'm wondering why the check for the env variable cannot happen in PackageIdValidator.cs, or why it's more desirable to create a new class here.